### DISCUSSION TITLE
Manifest V3

### DISCUSSION THREAD

**Discussion by RoyRao2333** [Alex313031/Thorium-MacOS#50] [General]
https://developer.chrome.com/blog/resuming-the-transition-to-mv3

So is there any chance to continue supporting MV2 in the future releases of Thorium?

Or is there any workaround to deal with it since half of my extensions are MV2?

**Comment by RoyRao2333** [comment]
https://github.com/Alex313031/thorium/discussions/12

**Comment by gz83** [comment]
We will temporarily use enterprise policies to extend support, but this support may only last for a year at most.

Also, is MV3 really as bad as the media claims? I see that many of the problems in MV3 have been solved.

**Reply by RoyRao2333** [reply]
> is MV3 really as bad as the media claims?

Don't know about this but I've tried [AdGuard MV3](https://github.com/AdguardTeam/AdGuardMV3), however, it does not filter as many ads as the MV2 version, and its functions are also very simple.

This could be better though, as the maintainers of these extensions will continue upgrading them.

Anyway, Thorium could be the best Chromium fork I've ever used. Thanks for the efforts!

**Comment by Alex313031** [comment]
@RoyRao2333 @gz83 @insign @mzinner @midzer Yes, MV3 is as bad as it seems. uBlock Origin Lite (it is called light for a reason), is severely crippled, and doesn't block nearly as many ads. There are also many great MV2 extensions that either cant work via MV3 at all, or havent been updated to MV3 (and although I know how to do that, and have done it for a few extensions I use alot, I can't be bothered to be the saviour of all extensions and devote all my time to migrating every extension to MV3 LOL).

The plan is to extend support via enterprise policies, and when they finally actually remove the MV2 code, to revert and restore that code. Maybe even possibly make an alternative web store on my website with the most popular MV2 extensions, and change the internal permissions to allow .crx installation from that hypothetical site.

In any case, I wouldn't want to use my own browser if it comes down to not having MV2 support, so this is a priority for me, and when the time comes to revert the code, I will do it even if it is alot of work and involves a bunch of files.

**Reply by RoyRao2333** [reply]
Thanks Alex, truly appreciated it since this is the best Chromium fork I've ever used. :)

**Reply by mzinner** [reply]
I was deeply worried about this as well. @Alex313031 excellent plan as usual, hope this does not turn out to be a huge amount of work for you.
